framework/tst/dslabs/framework/testing/newviz/SingleNodePanel.java

```diff
@@ -147,14 +160,32 @@ void updateState(EventTreeState s, boolean viewDeliveredMessages) {
             if (messages.containsKey(message)) {
                 continue;
             }
-            addMessage(message);
+            addMessage(message,
+                       !searchDone && (settings == null || settings.shouldDeliver(message)));
             repaintMessageBox = true;
         }
         for (MessageEnvelope message : new HashSet<>(messages.keySet())) {
             if (!ms.contains(message)) {
                 messageBox.remove(messages.get(message).getLeft());
                 repaintMessageBox = true;
                 messages.remove(message);
+            } else {
+                // Update messages if they are not deliverable. Slightly gross because we have to
+                // get the components of the message panel and search for the JButton.
+                JPanel mbox = messages.get(message).getLeft();
+                synchronized (mbox.getTreeLock()) {
```

Comment:
Ah, okay. I looked at the JavaDoc. I’m not a Swing expert yet, but I don’t think this is strictly necessary. We’re doing all of our updating on the Event Dispatch thread. We already have the lock or it’s not needed I think. That said, all of this is moot because it’s just nicer to store a reference to the button and use that.

Reply:
Fixed in https://github.com/emichael/dslabs/pull/29/commits/472418acfffa20d1d968e3e5a293784bd37e2bb0.